gramps/gen/db/upgrade.py

```diff
@@ -107,7 +121,14 @@ def gramps_upgrade_21(self):
             # has "handle", "_class", and uses json.dumps()
             self._commit_raw(json_data, key)
             self.update()
+        # DROP COLUMN is available with Sqlite v 3.35.0, released 2021-03-12
+        summary = self.get_summary()
+        db_ver = summary["Database version"].split(".")
+        if int(db_ver[0]) == 3 and int(db_ver[1]) >= 35:
+            self.dbapi.execute("ALTER TABLE %s DROP COLUMN blob_data;" % table_name)
```

Comment:
We should turn that into a DB API method (need to create) so that we can have versions for mongodb, postgresql, or whatever.

```suggestion
            self.dbapi.drop_column(table_name, "blob_data")
```

Reply:
We can't assume that we are upgrading a DBAPI database here.  The `self.dbapi` variable may not exist.

Perhaps we should keep the `gen/db/upgrade` module for object format upgrades and make backend specific changes elsewhere.